### PR TITLE
Deguice a few classes in IndicesModule

### PR DESCRIPTION
**Pull request by nik9000**
UpdateHelper, MetaDataIndexUpgradeService, and some recovery
stuff.

Move a test from a single node test to a unit test.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/transport/TransportService.java [hunk 1]
```diff
@@ -159,8 +157,9 @@ protected TaskManager createTaskManager() {
         return new TaskManager(settings);
     }
 
-    // These need to be optional as they don't exist in the context of a transport client
-    @Inject(optional = true)
+    /**
+     * Register this service's support for dynamic settings from the {@link ClusterSettings}. This isn't available to the transport client.
+     */
     public void setDynamicSettings(ClusterSettings clusterSettings) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: Why not pass ClusterSettings to the ctor?

**Review by rjernst — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/transport/TransportService.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: Yeah, I'll just move it there and make it nullable.

**Comment by nik9000** [comment]
Master: cf4038b66870331838832a5df9a8ccd1609580cd
5.x: afa9c6242b5857bcbe28544e9f9cdef64eba000e

**Pull request merged**